### PR TITLE
Prevent stalling on commands returning pairs

### PR DESCRIPTION
**Pull request by zedeus**
When nothing is returned readLine would stall forever. To check this we see if the reply is of kind `replyOk`, in which case we must quit early.

### CODE REVIEW

File: src/mpdclient.nim [hunk 1]
```diff
@@ -301,11 +301,17 @@ proc parseReply(s: string): Reply =
     let splits = s.split(':', maxsplit = 1)
     Reply(kind: replyPair, key: splits[0], value: splits[1].strip)
 
-template expectOk(mpd: MPDClient) =
+proc expectOk(mpd: MPDClient) =
   let reply = mpd.readLine.parseReply
   if reply.kind == replyAck:
     raise newException(CatchableError, reply.ack)
 
+proc expectPair(mpd: MPDClient): bool =
```
Comment (SolitudeSF) [inline src/mpdclient.nim]: why does this return `bool`?

File: src/mpdclient.nim [hunk 2]
```diff
@@ -301,11 +301,17 @@ proc parseReply(s: string): Reply =
     let splits = s.split(':', maxsplit = 1)
     Reply(kind: replyPair, key: splits[0], value: splits[1].strip)
 
-template expectOk(mpd: MPDClient) =
+proc expectOk(mpd: MPDClient) =
```
Comment (SolitudeSF) [inline src/mpdclient.nim]: i dont know if it gets inlined with `-O3` so please `{.inline.}`.

File: src/mpdclient.nim (hunk 1 above)
Comment (SolitudeSF) [reply src/mpdclient.nim]: ok, i get it

Comment (SolitudeSF) [reply src/mpdclient.nim]: this should be `expectList` and inline it too for good measure

File: src/mpdclient.nim [hunk 3]
```diff
@@ -626,6 +632,10 @@ proc runCommandOk(mpd: MPDClient; cmd: string, args: varargs[string]) {.inline.}
   mpd.runCommand(cmd, args)
   mpd.expectOk
 
+proc runCommandPair(mpd: MPDClient; cmd: string, args: varargs[string]): bool {.inline.} =
```
Comment (SolitudeSF) [inline src/mpdclient.nim]: `runCommandList`

**Review by SolitudeSF — CHANGES_REQUESTED** [review]

File: src/mpdclient.nim (hunk 1 above)
Comment (zedeus) [reply src/mpdclient.nim]: In `expectOk` we expect `replyOk`, so to me `expectPair` makes more sense since we want `replyPair`, I'll change it

Comment (SolitudeSF) [reply src/mpdclient.nim]: i want to leave the name open for if we need strict expectPair

**Pull request merged**